python/lsst/resources/gs.py

```diff
@@ -0,0 +1,279 @@
+# This file is part of lsst-resources.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# Use of this source code is governed by a 3-clause BSD-style
+# license that can be found in the LICENSE file.
+
+"""Accessing Google Cloud Storage resources."""
+
+from __future__ import annotations
+
+__all__ = ("GSResourcePath",)
+
+import contextlib
+import logging
+import re
+import tempfile
+from typing import IO, TYPE_CHECKING, Iterator, List, Optional, Set, Tuple, Union
+
+from google.cloud import exceptions, storage
```

Comment:
Yes. That is true for the S3 and HTTP implementations. It seemed easier than adding protection in every method but in this case it might work to put the `raise ImportError` in the `.client` property method since we always go through that. It's why pipe_base depends on boto3 (which is itself a bug because that functionality should be in ResourcePath itself).

Reply:
Since you conditionally import `gs.py`, can't you have a `try`/`except` block around this import and raise a nicer/more explanatory exception?